src/logging/request-obfuscator.js

```diff
@@ -0,0 +1,101 @@
+
+/**
+ * Module dependencies.
+ */
+
+import _ from 'lodash';
+import methods from '../methods';
+
+/**
+ * Get method by its lowerCase.
+ */
+
+function getMethod(method) {
+  return methods[_.findKey(methods, (object, key) => {
+    return key.toLowerCase() === method.toLowerCase();
+  })];
+}
+
+/**
+ * Obfuscate the response body.
+ */
+
+function obfuscateResponseBody(body, methodName) {
+  const method = _.get(getMethod(methodName), 'obfuscate.response');
+
+  if (!method || _.isEmpty(body.result)) {
+    return body;
+  }
+
+  body.result = method(body.result);
+
+  return body;
+}
+
+/**
+ * Obfuscate the response.
+ */
+
+function obfuscateResponse(request, instance) {
+  const requestBody = JSON.parse(instance.body);
+
+  if (_.isArray(request.response.body)) {
+    const requestMethods = _.map(requestBody, result => result.method);
+
+    request.response.body = _.map(request.response.body, (body, index) => obfuscateResponseBody(body, requestMethods[index]));
```

Comment:
It looks like you're assuming that the response is ordered, which means you're ignoring the fact that there is an id to make that correlation instead.